sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/FlowAnalysis/ControlFlowGraphTest.cs

```diff
@@ -1483,133 +1483,200 @@ public void Cfg_TryFinally()
+                baz();
             }
             finally 
             {
-                foo();
+                cw1();
             }
+            cw2();
             ");
 
-            VerifyCfg(cfg, 7);
+            VerifyCfg(cfg, 8);
 
             var blocks = cfg.Blocks.ToList();
 
-            var tryBlock = blocks[0];
+            var tryStartBlock = blocks[0];
             var binaryBlock = blocks[1];
             var whenNullBlock = blocks[2];
             var whenNotNullBlock = blocks[3];
             var assignmentBlock = blocks[4];
             var finallyBlock = blocks[5];
-            var exitBlock = blocks[6];
+            var afterFinallyBlock = blocks[6];
+            var exit = blocks.Last();
+
+            tryStartBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { binaryBlock });
+            VerifyAllInstructions(tryStartBlock, "cw0", "cw0()");
 
-            tryBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { binaryBlock });
             binaryBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { whenNullBlock, whenNotNullBlock });
+            VerifyAllInstructions(binaryBlock, "e", "null", "e == null");
+
             whenNullBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { assignmentBlock });
+            VerifyAllInstructions(whenNullBlock, "null");
+
             whenNotNullBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { assignmentBlock });
+            VerifyAllInstructions(whenNotNullBlock, "e");
+
             assignmentBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { finallyBlock });
-            finallyBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { exitBlock });
+            VerifyAllInstructions(assignmentBlock, "message = e == null ? null : e", "baz", "baz()");
 
-            exitBlock.Should().BeOfType<ExitBlock>();
+            finallyBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { afterFinallyBlock, exit });
+            VerifyAllInstructions(finallyBlock, "cw1", "cw1()");
+
+            afterFinallyBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { blocks.Last() });
+            VerifyAllInstructions(afterFinallyBlock, "cw2", "cw2()");
+
+            blocks.Last().Should().BeOfType<ExitBlock>();
         }
 
         [TestMethod]
         [TestCategory("CFG")]
-        public void Cfg_TryCatch()
+        public void Cfg_TryCatchFinally()
         {
             var cfg = Build(@"
             cw0();
             try
             {
                 cw1();
             }
-            catch
+            catch(Exception1 e)
             {
                 cw2();
             }
-            cw3();");
+            catch(Exception2 e)
+            {
+                cw3();
+            }
+            finally
+            {
+                cw4();
+            }
+            cw5();");
 
-            VerifyCfg(cfg, 4);
+            VerifyCfg(cfg, 7);
 
             var blocks = cfg.Blocks.ToList();
 
             var tryStartBlock = blocks[0];

```

Comment:
As suggested above - perhaps do that?
var beforeTryBlock = blocks[0];
var tryBlock = blocks[1];